pkg/cli/image/manifest/manifest.go

```diff
@@ -279,21 +280,30 @@ func FirstManifest(ctx context.Context, from imagereference.DockerImageReference
 	}
 
 	originalSrcDigest := srcDigest
-	srcManifests, srcManifest, srcDigest, err := ProcessManifestList(ctx, srcDigest, srcManifest, manifests, from, filterFn, false)
+	// Do not return a manifest list, filter here to return appropriate os/arch
+	if o.IsWildcardFilter() {
+		// srcManifest returned to print all os/arch options with error message
+		return srcManifest, ManifestLocation{}, &WildcardFilterErr{}
+	}
+	srcManifests, srcManifest, srcDigest, err := ProcessManifestList(ctx, srcDigest, srcManifest, manifests, from, o.FilterByOS, o.Include, false)
 	if err != nil {
 		return nil, ManifestLocation{}, err
 	}
-	if len(srcManifests) == 0 {
-		return nil, ManifestLocation{}, fmt.Errorf("filtered all images from manifest list")
+	// Should only ever return 1 manifest from here
+	m := len(srcManifests)
+	switch {
+	case m > 1:
+		return nil, ManifestLocation{}, fmt.Errorf("multiple manifests match platform %s, failed to choose single manifest", o.FilterByOS)
```

Comment:
with the fix below of removing dupes from filtered if/only if filterByOS == manifest.Platform, i'm not sure the returning multiple manifests is possible anymore but in case, better to differentiate. It would be if an image had duplicate digests for same platform os/arch (don't know if there's anything preventing that w/ image builds)

Reply:
still not sure an image can't have multiple platforms per digest, so leaving this check here since otherwise the error will be misleading.